src/channel_server/player_data_provider.cpp

```diff
@@ -235,7 +235,9 @@ auto player_data_provider::stop_following(ref_ptr<player> follower) -> void {
 }
 
 auto player_data_provider::disconnect() -> void {
-	for (auto &kvp : m_players) {
+	auto m_players_copy = hash_map<game_player_id, ref_ptr<player>>(m_players);
```

Comment:
Does that really make a raw copy of it? it sounds like its just storing the reference, to me